src/tools/cephfs_mirror/PeerReplayer.cc

```diff
@@ -1049,6 +1067,33 @@ void PeerReplayer::run(SnapshotReplayerThread *replayer) {
 }
 
 void PeerReplayer::peer_status(Formatter *f) {
+  std::scoped_lock locker(m_lock);
+  f->open_object_section("stats");
+  for (auto &[dir_path, sync_stat] : m_snap_sync_stats) {
+    f->open_object_section(dir_path);
+    if (!sync_stat.current_syncing_snap) {
+      f->dump_string("state", "idle");
+    } else {
+      f->dump_string("state", "syncing");
+      f->open_object_section("current_sycning_snap");
+      f->dump_unsigned("id", (*sync_stat.current_syncing_snap).first);
+      f->dump_string("name", (*sync_stat.current_syncing_snap).second);
+      f->close_section();
+    }
+    if (sync_stat.last_synced_snap) {
+      f->open_object_section("last_synced_snap");
+      f->dump_unsigned("id", (*sync_stat.last_synced_snap).first);
+      f->dump_string("name", (*sync_stat.last_synced_snap).second);
+      if (sync_stat.last_sync_duration) {
+        f->dump_float("sync_duration", *sync_stat.last_sync_duration);
+      }
+      f->close_section();
+    }
+    f->dump_unsigned("snaps_synced", sync_stat.synced_snap_count);
+    f->dump_unsigned("snaps_deleted", sync_stat.deleted_snap_count);
```

Comment:
snaps renamed ?

Reply:
I thought about that, but failed to infer how would that be useful. Snaps synced and deleted are particularly useful, but would snaps renamed be of much use?

Reply:
I tend to think that stats are for audit purpose.
You may want to go through the rename stats to confirm that your actions have been accounted for ... no more and no less.

Reply:
Ummm... fair point. I'll include that too. Thx!